### PR TITLE
Pecan cookies

### PR DESCRIPTION
**Pull request by javierarce**
This PR introduces pecan to CartoDB Editor. It's enabled when both `pecan_cookies` and `new_modals` flags are `true`.

The main code is here: https://github.com/CartoDB/cartodb/blob/3336-pecan/lib/assets/javascripts/cartodb/common/dialogs/pecan/pecan_view.js

This dialog is called from mapview.js: https://github.com/CartoDB/cartodb/pull/4250/files#diff-efdf441d0e50632831fb15d9edd1c982R660

:fire: No specs yet.

### CODE REVIEW

File: lib/assets/javascripts/cartodb/common/dialogs/pecan/pecan_view.js [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+    this.close();
+    //this.trigger("skip", this); // TODO: enable this after the manual testing is finished
+  },
+
+  _onLoadWizard: function() {
+    this.layer.wizard_properties.set({ property: this.model.get("response").column, metadata: this.model.get("response").metadata });
+  },
+
+  _analyzeColumns: function() {
+    this.columns.each(this._analyzeColumn);
```
Comment (javisantana) [inline lib/assets/javascripts/cartodb/common/dialogs/pecan/pecan_view.js]: I'd add a filter here for typical column names (it does not need to be done before merging, it's an improvement.

https://github.com/CartoDB/cartodb.js/blob/describe/test_describe.html#L211

cc @andrewxhill

File: app/models/user.rb [hunk 2]
```diff
@@ -2026,7 +2026,7 @@ def load_cartodb_functions(statement_timeout = nil, cdb_extension_target_version
   # Upgrade the cartodb postgresql extension
   def upgrade_cartodb_postgres_extension(statement_timeout=nil, cdb_extension_target_version=nil)
     if cdb_extension_target_version.nil?
-      cdb_extension_target_version = '0.7.3'
+      cdb_extension_target_version = '0.8.0'
```
Comment (javisantana) [inline app/models/user.rb]: just to be totally sure, if the user has 0.7.3 in his database, cartodb will work well (except pecan which needs 0.8.0), right?

cc @rafatower

File: lib/assets/javascripts/cartodb/common/dialogs/pecan/pecan_view.js (hunk 1 above)
Comment (andrewxhill) [inline lib/assets/javascripts/cartodb/common/dialogs/pecan/pecan_view.js]: Here is the list we came up with so far

https://github.com/CartoDB/cartodb.js/blob/describe/test_describe.html#L211

On Wed, Jul 1, 2015 at 10:29 AM, javi santana <notifications@github.com>
wrote:

> In lib/assets/javascripts/cartodb/common/dialogs/pecan/pecan_view.js
> <https://github.com/CartoDB/cartodb/pull/4250#discussion_r33683463>:
>
> > +    if (this.layer) {
> > +      this.layer.wizard_properties.unbind("load", this._onLoadWizard, this);
> > +      this.layer.wizard_properties.bind("load", this._onLoadWizard, this);
> > +      this.layer.wizard_properties.active(this.model.get("response").wizard);
> > +    }
> > +
> > +    this.close();
> > +    //this.trigger("skip", this); // TODO: enable this after the manual testing is finished
> > +  },
> > +
> > +  _onLoadWizard: function() {
> > +    this.layer.wizard_properties.set({ property: this.model.get("response").column, metadata: this.model.get("response").metadata });
> > +  },
> > +
> > +  _analyzeColumns: function() {
> > +    this.columns.each(this._analyzeColumn);
>
> I'd add a filter here for typical column names (it does not need to be
> done before merging, it's an improvement.
>
> https://github.com/CartoDB/cartodb.js/blob/describe/test_describe.html#L211
>
> cc @andrewxhill <https://github.com/andrewxhill>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CartoDB/cartodb/pull/4250/files#r33683463>.
>



-- 

andrewxhill <http://twitter.com/andrewxhill>
http://cartodb.com

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4663/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4664/))

File: app/models/user.rb (hunk 2 above)
Comment (rafatower) [inline app/models/user.rb]: right

File: lib/assets/javascripts/cartodb/common/dialogs/pecan/pecan_view.js (hunk 1 above)
Comment (saleiva) [inline lib/assets/javascripts/cartodb/common/dialogs/pecan/pecan_view.js]: I've a few more (created_at, updated_at, iso2, iso3, x_coord, y_coord, xcoord, ycoord)

Comment (fdansv) [inline lib/assets/javascripts/cartodb/common/dialogs/pecan/pecan_view.js]: :white_check_mark:

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4667/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4670/))

**Pull request merged**